### DISCUSSION TITLE
Please integrate with AI in such a way that it helps to generate sql queries easily for users and make this project more user friendly

### DISCUSSION THREAD

**Discussion by Greatz08** [apache/superset#27272] [Ideas]
I know we can use chatgpt or gemini or groq or huggingface or ollama with open source model self hosted i mean there are many ways to use ai and generate sql queries for our need and try in superset to convert it to graph.With Rag we can feed database and ask questions or ask sql queries for different things so it is possible BUT if team superset can create or use any open source model which is lightweight and just for sql queries only that can be integrated within superset so that where ever there is context menu for writing sql queries besides that or in that we can use that model to instruct what we want for our data and it reads our data and generate sql query for that purpose and then we can create graph with superset and manipulate whatever we want accordingly.You can make this feature optional to initially but this will be very helpful and i believe can be better than general purpose ai models plus can be used for free without any censorship and our data wont be leaked to those ai servers will be big plus for open source community.Many will be able to adopt this awesome solution easily without having to worry too much about sql language and in this way it can become more user friendly too. I hope my idea and explanation is satisfactory for this implementation.Rest i am open to this discussion further if needed but please do consider this will be big thing for this awesome project.

**Comment by surapuramakhil** [comment]
https://github.com/apache/superset/issues/28167
